src/app/models/evt-models.ts

```diff
@@ -15,3 +15,50 @@ export interface PageData {
     originalContent: OriginalEncodingNodeType[];
     parsedContent: GenericElementData[];
 }
+
+export type AttributesData = Array<{ key: string; value: string }>;
+
+export interface OriginalEncodingData {
+    originalEncoding: OriginalEncodingNodeType;
+}
+
+export type NamedEntityType = 'person' | 'place' | 'org' | 'relation' | 'event' | 'generic';
+export interface NamedEntitiesList extends GenericElementData {
+    id: string;
+    label: GenericElementData | string;
+    namedEntityType: NamedEntityType;
+    icon?: string;
+    description?: Description;
+    sublists: NamedEntitiesList[];
+    content: NamedEntity[];
+    relations: Relation[];
+    originalEncoding: OriginalEncodingNodeType;
+}
+
+export interface NamedEntity extends GenericElementData {
+    id: string;
+    label: NamedEntityLabel;
+    namedEntityType: NamedEntityType | 'personGrp';
+    content: NamedEntityInfo[];
+    occurrences: string[]; // TODO: evaluate which type assign
+    originalEncoding: OriginalEncodingNodeType;
+}
+
+export type NamedEntityLabel = GenericElementData | string | Array<GenericElementData | string>;
```

Comment:
A label should be a string. When are other types needed?

Reply:
I have probably over-complicated the situation. I was try to handle all the different type of xml encoding we can face when using TEI encoded files. Details of an entity, such as `<person>` can be encoded in many different ways, both structured and unstructured; it can even be defined as an empty element. E.g.:
```
<person xml:id="zh-tw_LB"/>

<person xml:id="pp1">
   Person pp1
</person>

<person xml:id="cwn">
   <p>Customer WN</p>
</person> 

<person xml:id="Acaritus">
   <persName>
      <forename>Acaritus</forename>
   </persName>
   <sex>M</sex>
</person>

<person xml:id="AAAAATEST">
   <persName>
      <forename>Edward</forename>
      <forename>George</forename>
      <surname type="linked">Bulwer-Lytton</surname>,
      <roleName>Baron Lytton of
         <placeName>Knebworth</placeName>
      </roleName>
   </persName>
   <sex>M</sex>
   <occupation>syndicus et procurator communium et universitatum de Marola et de
      Matrono</occupation>
</person>
```
But you are right: the label is the text used to identify the entity, thus it should probably be a simple string, and I can handle all the possible cases differently, by just extracting the `textContent`. 

This make me think that property of entities (such as information encoded in attributes and child nodes) should be more defined, maybe by just using possible child nodes from `namesdates` module for the moment.